cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1240,22 +1240,22 @@ MultiSymbolReadOutput LocalVersionedEngine::batch_read_and_join_internal(
     }
     auto clauses_ptr = std::make_shared<std::vector<std::shared_ptr<Clause>>>(std::move(clauses));
     return folly::collect(symbol_processing_result_futs).via(&async::io_executor())
-    .thenValueInline([this, &handler_data, clauses_ptr, component_manager](std::vector<SymbolProcessingResult>&& symbol_processing_results) mutable {
+    .thenValueInline([this, &handler_data, clauses_ptr, component_manager, &read_options](std::vector<SymbolProcessingResult>&& symbol_processing_results) mutable {
```

Comment:
Capture `read_options` by value here and in other placed in this function. The reason is that any of the of the functions can throw and we're using `.get` on the future which just propagates the exception. But there might be other threads working on the data. The stack will get cleaned though and `read_options` will be destroyed and then the other treads will segfault. Read options holds only a shared_ptr so it's cheap to copy.